content/en/docs/Core concepts/record_implementation.md

```diff
@@ -0,0 +1,202 @@
+100 extra fields can be stored at runtime in a `LogLine` (also hardcoded with `NumFieldsBaker`),
+these extra fields are a fast way to exchange data between filters and/or outputs but they are
+neither handled during Parsing (i.e `LogLine.Parse`) nor serialization (`LogLine.ToText`).
+
+### Custom LogLine
+If the hardcoded values for `LogLineNumFields` and `NumFieldsBaker` do not suit your needs,
+it's advised that you copy `LogLine.go` in your project and modify the constants declared at
+the top of the file. Your specialized `LogLine` will still implement `baker.Record` and thus
+can be used in lieu of `baker.LogLine`. To do so, you need to provide a `CreateRecord`
+function to `baker.Components` when calling `baker.NewConfigFromToml`.
+
+For example (`my.LogLine` is your custom implementation):
+
+```go
+comp := baker.Components{}
+
+comp.CreateRecord = func() baker.Record {
+  return &my.LogLine{ FieldSeparator: ',' }
+}
+```
+
+## How to use a custom version of the Record
+
+Once a customized version of a Record has been implemented, you want to use it in your code.
+In order to do so, some functions may be implemented while instantiating `baker.Components`:
+
+```go
+type Components struct {
+	Validate      ValidationFunc
+    CreateRecord  func() Record
+    FieldByName func(string) (FieldIndex, bool)
+    FieldName   func(FieldIndex) string
+    //... other functions
+}
+```
+
+### Validate
+
+`Validate` is the function used to validate a record. It is called for each processed record
+unless not set or when the `[general] dont_validate_fields = true` configuration is set in
+the TOML file.
+
+Regardless of the TOML configuration, the function is passed to all components that can use
+it at their will.
+
+### CreateRecord
+
+`CreateRecord` is the function that creates a new record. If not set, a default function is
+used that creates a `LogLine` with `,` as field separator.
+
+The function is used internally by Baker to create new records every time a new one comes from
+the input.
+
+The function is also passed to components that can use it to create new records while processing.
+
+### FieldByName
+
+`FieldByName` gets a field index by its name. The function is mainly used by the components
+(that receive it during setup) to retrieve the index of a field they need for filtering or
+processing, but it is also used internally by Baker when sending fields to the output
+(when at least one field is selected in the output TOML configuration).
+
+### FieldIndex
+
+`FieldName` gets a field name by its index. The function is passed to components that can use
+it for their internal logic.
+
+## RecordConformanceTest
+
+The `test_helper.go` provides a `RecordConformanceTest` test helper whose goal is to give the
+user a structured test for new implementations of the Record.
+
+The helper receives the implementation of `CreateRecord` and creates new records testing
+them against a set of requirements.
```

Comment:
I'd add  or include the fact that the conformance test provides a way to verify that a record implementation respects the invariant that Baker requires for a Record implementation